shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -705,7 +705,7 @@ protected Bundle createArgs() {
   // Delegate that runs all lifecycle and OS hook logic that is common between
   // FlutterActivity and FlutterFragment. See the FlutterActivityAndFragmentDelegate
   // implementation for details about why it exists.
-  @VisibleForTesting @Nullable /* package */ FlutterActivityAndFragmentDelegate delegate;
+  @VisibleForTesting @Nullable protected FlutterActivityAndFragmentDelegate delegate;
```

Comment:
See comment https://github.com/flutter/flutter/issues/93905#issuecomment-989050277

I don't think this is the right fix

Reply:
In the shared engine scenario (eg, [flutter_boost](https://github.com/alibaba/flutter_boost)), We expect to be able to take over ActivityControlSurface in the FlutterActivity/FlutterFragment subclass(eg, call `ActivityControlSurface.attachToActivity`), so here make ExclusiveAppComponent visibile to FlutterActivity/FlutterFragment subclass.

```java
  void attachToActivity(
      @NonNull ExclusiveAppComponent<Activity> exclusiveActivity, @NonNull Lifecycle lifecycle);
```

Reply:
As stated in the comments https://github.com/flutter/flutter/issues/93905#issuecomment-989673591, We need this PR.

Reply:
Can we instead expose the method we actually need to subclasses.

Reply:
> Can we instead expose the method we actually need to subclasses.

Thank you for your reply.

In this case, I think exposing the `delegate` of `FlutterFragment` as a `protected` member to subclasses is the least costly solution.